src/config/loading/mod.rs

```diff
@@ -151,6 +153,25 @@ pub async fn load_from_paths_with_provider_and_secrets(
         load_builder_from_paths(config_paths)?
     };
 
+    // Check secrets in configuration
+    #[cfg(feature = "enterprise")]
+    {
+        let (source, _) = crate::config::load_source_from_paths(config_paths)?;
+
+        let json = crate::config::util::json::serialize(source, &builder, false, false);
+        let json = json.expect("config should be serializable");
```

Comment:
Might make sense to move this loading into the block below so that we only do it if the enterprise mode is actually being used.